Replicator/c4RemoteReplicator.hh

```diff
@@ -221,14 +221,20 @@ namespace c4Internal {
             unsigned delay = 1 << std::min(retryCount, 30u);
             unsigned maxDelay = getIntProperty(kC4ReplicatorOptionMaxRetryInterval,
                                                kDefaultMaxRetryDelay);
+            maxDelay = getIntProperty(kC4ReplicatorOptionMaxAttemptWaitTime, maxDelay);
+            if (maxDelay == 0) {
+                maxDelay = kDefaultMaxRetryDelay;
+            }
             return std::min(delay, maxDelay);
         }
 
 
         // Returns the maximum number of (failed) retry attempts.
         unsigned maxRetryCount() const {
             unsigned defaultCount = continuous() ? kMaxContinuousRetryCount : kMaxOneShotRetryCount;
-            return getIntProperty(kC4ReplicatorOptionMaxRetries, defaultCount);
+            unsigned ret = getIntProperty(kC4ReplicatorOptionMaxRetries, defaultCount);
+            ret = getIntProperty(kC4ReplicatorOptionMaxAttempts, ret + 1);
+            return ret == 0 ? defaultCount : ret - 1;
```

Comment:
Either very clever or a happy accident that a continuous replicator with neither of these properties set will end up as 0 due to overflow!

Reply:
Do you mean the situation when (ret + 1) overflow?

Reply:
Good catch!